src/ol/renderer/canvas/canvasvectorlayerrenderer.js

```diff
@@ -194,6 +227,14 @@ ol.renderer.canvas.VectorLayer.prototype.prepareFrame =
 
   var extent = ol.extent.buffer(frameStateExtent,
       vectorLayerRenderBuffer * resolution);
+  var projectionExtent = viewState.projection.getExtent();
+
+  if (vectorSource.getWrapX() && viewState.projection.isGlobal() &&
```

Comment:
We also require extents for non-global projections (for tile grid creation), but we don't want to wrap them.